ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -191,10 +28,36 @@ def add_arguments(self, parser):
             "The default of -1 considers ALL documents.",
         )
 
+    def iter_solr(self, nsize=10, item_type='page'):
+        i=0
+        q=SolrQuerySet().search(item_type=item_type)
+        total = q.count()
+        for i in progressbar(range(0, total, nsize)):
+            q.set_limits(i,i+nsize)
+            yield from q
+
+    def iter_pages(self):
+        for d in self.iter_solr(item_type='page'):
+            yield {k:v for k,v in d.items() if k in self.PAGE_OUTPUT_FIELDS}
```

Comment:
let Solr do this work for you - in parasolr queryset the method is `only` (inspired by django queryset) https://parasolr.readthedocs.io/en/latest/codedocs.html#parasolr.query.queryset.SolrQuerySet.only